lib/linguist/languages.yml

```diff
@@ -3601,6 +3601,18 @@ Python traceback:
   tm_scope: text.python.traceback
   ace_mode: text
   language_id: 304
+Q:
+  type: programming
+  color: "#0040CD"
+  aliases:
+  - Kdb+
+  extensions:
+  - ".q"
```

Comment:
https://github.com/github/linguist/pull/3638 attempted to include support for the `.k` extension too, which would have posed a problem due to clashing with non-Kdb, yet undefined, languages. This PR doesn't mention that extension at all. Whilst this isn't a problem, is `.q` the preferred extension for Q/Kdb+?

I notice that https://github.com/KxSystems/kdb which you reference includes `.k` files which won't be recognised as Q/Kdb+ once this PR has been merged, hence the question.

Reply:
i have removed the extra new line, and added two more samples.  the `.k` samples from #3638 are not needed because we are not looking for the `.k` extension to be formally recognized in linguist. `.q` is the proper extension for Q/Kdb+.  in fact, users can use Q syntax highlighting to markup their .k files as well, so this PR will allow that - just not automatically.

Reply:
> in fact, users can use Q syntax highlighting to markup their .k files as well

Since k is accessible from q in `k)` prefixed lines, a good q syntax highlighter should know how to markup k code, but the k language is sufficiently different from q that using a q highlighter on k files is unlikely to produce good results.  The differences include over 100 q keywords that are not special in k and the treatment of `_` character in identifiers.

Reply:
Q.tmbundle inherits from K4.tmbundle. so my suggestion is that people can update they .gitattributes file to link the .k extension to the K4 syntax. i agree that it doesn’t handle the _ difference between k and q.

Reply:
People will not be able to link to K4 syntax unless K4 language is known to Linguist.  It is frustrating that the dispute over default extensions prevents the Linguist from learning about the Q and K languages.  Personally, I would not care if the default extensions were `.kxq` and `.kxk`.  I would still be able to configure my repositories to use `.q` and `.k`.  However this will only be possible once `languages.yml` has two new entries.

Yes, having just Q will be a progress, but let's not give an impression that Q and K are the same language.